### PR TITLE
Added initial GUI

### PR DESCRIPTION
**Pull request by spowelljr**
v0.0.1 of the minikube GUI

Screenshots on Linux
Screen Shot 2022-03-03 at 9 37 44 AM
Screen Shot 2022-03-03 at 9 38 15 AM
Screen Shot 2022-03-03 at 9 38 30 AM

**GUI Binaries**
Mac: https://storage.googleapis.com/minikube-gui/v0.0.1/minikube-gui-mac.zip
Linux: https://storage.googleapis.com/minikube-gui/v0.0.1/minikube-gui-linux.zip
Windows: https://storage.googleapis.com/minikube-gui/v0.0.1/minikube-gui-windows.zip

### CODE REVIEW

File: gui/LICENSE [hunk 1]
```diff
@@ -0,0 +1,27 @@
+Copyright (C) 2016 The Qt Company Ltd.
```
Comment (afbjorklund) [inline gui/LICENSE:1]: I think this one should have a minikube Copyright at the top of it.

File: gui/README.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+$ make
+...
+$ ./systray
+```
+
+----
+
+See <https://doc.qt.io/qt-5/qtwidgets-desktop-systray-example.html>
+
+minikube logo from the <https://github.com/kubernetes/minikube> project.
```
Comment (afbjorklund) [inline gui/README.md:24]: this might be redundant, if the code is _in_ the minikube project ?

File: gui/instance.h [hunk 3]
```diff
@@ -0,0 +1,67 @@
+#ifndef INSTANCE_H
+#define INSTANCE_H
+
+#include <QAbstractListModel>
+#include <QString>
+#include <QList>
+#include <QMap>
+
+//! [0]
+class Instance
```
Comment (afbjorklund) [inline gui/instance.h:10]: You could rename this class to "Profile", or maybe better yet "Cluster", to match minikube vocabulary ?

File: gui/window.cpp [hunk 4]
```diff
@@ -0,0 +1,472 @@
...
+    QModelIndexList index = model->match(start, Qt::DisplayRole, instance);
+    if (index.size() == 0) {
+        return;
+    }
+    instanceListView->setCurrentIndex(index[0]);
+}
+
+void Window::createInstanceGroupBox()
+{
+    instanceGroupBox = new QGroupBox(tr("Instances"));
```
Comment (afbjorklund) [inline gui/window.cpp:295]: Maybe "Clusters", or something ? Same as `minikube profile list`

**Comment by afbjorklund** [comment]
Looking great! I think it should have some minor cleanups and overhaul, to "translate" some of the Lima things (like "instance")

You might want to consider adding `clang-format` or some other automatic .cpp cleanup tool, similar to the `gofmt` used with .go

**Comment by afbjorklund** [comment]
I would love to see a "SSH" button and a "Dashboard" button, opening up a terminal or a web browser - respectively

**Review by afbjorklund — CHANGES_REQUESTED** [review]
Please update the license, and possible change some class names.

**Review by afbjorklund — APPROVED** [review]
Looks good! Not sure if the license needs to be in every file, but thanks for updating it!

File: gui/window.cpp [hunk 5]
```diff
@@ -0,0 +1,587 @@
...
+{
+    QDialog dialog;
+    dialog.setWindowTitle(tr("Set Cluster Values"));
+    dialog.setModal(true);
+
+    QFormLayout form(&dialog);
+    driverComboBox = new QComboBox;
+    driverComboBox->addItem("docker");
+    driverComboBox->addItem("hyperv");
+    driverComboBox->addItem("hyperkit");
```
Comment (afbjorklund) [inline gui/window.cpp:478]: the "kvm2" driver is missing in this list, for Linux users.

Also "virtualbox" should probably be available, no ?

**Comment by afbjorklund** [comment]
@medyagh : I would still prefer to have this as a separate git repository

File: gui/window.cpp (hunk 5 above)
Comment (spowelljr) [reply gui/window.cpp:480]: Added all the drivers and gated OS specific ones

**Comment by medyagh** [comment]
> @medyagh : I would still prefer to have this as a separate git repository

sounds good once we have a stable release, we should be asking for a repo in kuberentes sigs

**Pull request merged**